### PR TITLE
Merge Train/Add to Build

### PR DESCRIPTION
**Pull request by liliu-z**
Knowhere has `Build`, `Train` and `Add` APIs, their semantic should be:
`Build`: build a index in a batch.
`Train`: train the index w/o adding data
`Add`: add data to indexes.

For DiskANN, we should not support `Train` & `Add` but only `Build`

### CODE REVIEW

**Review by foxspy — APPROVED** [review]
/lgtm

File: src/index/diskann/diskann.cc [hunk 1]
```diff
@@ -43,17 +43,17 @@ class DiskANNIndexNode : public IndexNode {
...
     Status
-    Build(const DataSet& dataset, const Config& cfg) override {
-        return Add(dataset, cfg);
-    }
+    Build(const DataSet& dataset, const Config& cfg) override;
 
     Status
     Train(const DataSet& dataset, const Config& cfg) override {
-        return Status::success;
+        return Status::not_implemented;
```
Comment (zhengbuqian) [inline src/index/diskann/diskann.cc:50]: not implemented error should be used when either the code is being developed and it not implemented yet; or the base class forces its subclasses to implement a specific method by providing a default impl to return not_implemented error. `not_implemented` methods should eventually be implemented either after development or by subclasses.

IIUC we never want the Train/Add methods of DiskANN to be invoked, not_implemented doesn't seem to be the correct error code. How about invalid_args or a new code Status::failed_precondition/not_supported?

Comment (liliu-z) [reply src/index/diskann/diskann.cc:50]: Good catch, free feel to make a PR adding that

**Pull request merged**